syntax = "proto3";
package acc_engineer;
import "rpc.proto";

message Echo
{
  option(rpc.cmd_id) = 0;

  message Request
  {
    string message = 1;
  }

  message Response
  {
    string message = 1;
  }
}

message AuthenticationTCP
{
  option(rpc.cmd_id) = 1;

  message Request
  {
    string driver_name = 1;
    string password = 2;
    uint64 driver_id = 3;
  }

  message Response
  {
    uint64 driver_id = 2;
    string ticket = 3;

    int64 error_code = 100;
    string error_message = 101;
  }
}

message AuthenticationUDP
{
  option(rpc.cmd_id) = 2;

  message Request
  {
    string driver_name = 1;
    string password = 2;
    uint64 driver_id = 3;
  }

  message Response
  {
    uint64 driver_id = 2;
    string ticket = 3;

    int64 error_code = 100;
    string error_message = 101;
  }
}

message Driver
{
  string driver_name = 1;
  uint64 driver_id = 2;
}

message Strategy
{
  float fuel = 1;
  int32 tyre_set = 2;
  string tyre_compound = 3;
  repeated float tyre_pressures = 4;
  int32 brake_pad = 5;
  bool repair_bodywork = 6;
  bool repair_suspension = 7;
}

message DriverUpdate
{
  option(rpc.cmd_id) = 3;
  option(rpc.no_reply) = true;

  message Request
  {
    repeated Driver drivers = 1;
    uint64 version = 2;
  }

  message Response
  {
  }
}

message DriverTelemetryUpdate
{
  option(rpc.cmd_id) = 4;
  option(rpc.no_reply) = true;

  message Request
  {
    Driver driver = 1;
  }

  message Response
  {

  }
}

message DriverStrategyUpdate
{
  option(rpc.cmd_id) = 5;

  message Request
  {
    Driver driver = 1;
    Strategy strategy = 2;
  }

  message Response
  {

  }
}